registry.go

```diff
@@ -108,3 +112,9 @@ type SignatureService interface {
 	// Put stores the signature for the provided digest.
 	Put(dgst digest.Digest, signatures ...[]byte) error
 }
+
+// CatalogService provides a way of retrieving the names of each of the repositories
+type CatalogService interface {
+	// Get retrieves repository names from the registry.
+	Get(n int, q string) (p []string, moreEntries bool, err error)
```

Comment:
I think squishing it down makes sense.  It always felt a bit strange creating the service with a single method.  The only case where I can think of where we'd want to add more methods would be some kind of future search functionality, and I'm not sure if that's on the roadmap.